packages/mattermost-redux/test/merge_objects.ts

```diff
@@ -1,20 +1,20 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-function isObject(obj) {
-    return obj && typeof obj === 'object' && !Array.isArray(obj);
+function isObject(obj: any): boolean {
+    return Boolean(obj && typeof obj === 'object' && !Array.isArray(obj));
 }
 
 // Returns the result of merging two objects. If a field is specified in both a and b, the value from b takes precedence
 // unless both values are objects in which case mergeObjects will be called recursively.
-export default function mergeObjects(a, b, path = '.') {
+export default function mergeObjects(a: any, b: any, path = '.') {
```

Comment:
This didn't quite work. We can at least validate that the inputs are records and that will type check. Can't think of a way to avoid fudging the the recursive call and return type with `any`, though. Typescript's recursive type support is limited. Closest I could come is https://stackoverflow.com/questions/47842266/recursive-types-in-typescript , but that didn't work here either.

Reply:
This seems like a typing nightmare the more I look at it. I tried a few other things, but none of them worked.

Part of me wants to just replace this with `lodash/merge` and be done with it since I don't know why we rolled our own version to begin with. I tried briefly and that did introduce some new type errors, but there's a chance it just identified existing ones since the errors are all in one place